File: src/main/java/com/mallang/auth/query/dao/MemberProfileDataDao.java
package com.mallang.auth.query.dao;

import com.mallang.auth.query.dao.support.MemberQuerySupport;
import com.mallang.auth.query.data.MemberProfileData;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Transactional(readOnly = true)
@Component
public class MemberProfileDataDao {

    private final MemberQuerySupport memberQuerySupport;

    public MemberProfileData find(Long memberId) {
        return MemberProfileData.from(memberQuerySupport.getById(memberId));
    }
}

File: src/main/kotlin/com/mallang/auth/query/dao/MemberProfileDataDao.kt
package com.mallang.auth.query.dao

import com.mallang.auth.query.dao.model.MemberProfileQueryModel
import com.mallang.auth.query.dao.support.MemberQuerySupport
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Transactional(readOnly = true)
@Component
class MemberProfileDataDao(
        private val memberQuerySupport: MemberQuerySupport
) {

    fun find(memberId: Long): MemberProfileQueryModel {
        return MemberProfileQueryModel.from(memberQuerySupport.getById(memberId))
    }
}
